sdk/trts/trts_veh.cpp

```diff
@@ -199,18 +200,30 @@ extern "C" __attribute__((regparm(1))) void continue_execution(sgx_exception_inf
 extern "C" void restore_xregs(uint8_t *buf);
 
 extern "C" void constant_time_apply_sgxstep_mitigation_and_continue_execution(sgx_exception_info_t *info,
-        uintptr_t ssa_aexnotify_addr, uintptr_t stack_tickle_pages, uintptr_t code_tickle_page, uintptr_t data_tickle_page, uintptr_t c3_byte_address);
+        uintptr_t ssa_aexnotify_addr, uintptr_t stack_tickle_pages, uintptr_t code_tickle_page, uintptr_t data_tickle_address, uintptr_t c3_byte_address);
 
+// constant time select based on given condtion
+static inline uint64_t cselect64(uint64_t pred, const uint64_t expected, uint64_t old_val, uint64_t new_val)
+{
+    __asm__("cmp %3, %1\n\t"
+            "cmove %2, %0"
+            : "+rm"(new_val)
+            : "rm"(pred), "rm"(old_val), "i"(expected));
```

Comment:
It has an error in debug build
trts_veh.cpp: In function ‘uint64_t cselect64(uint64_t, uint64_t, uint64_t, uint64_t)’:
trts_veh.cpp:211:56: error: asm operand 3 probably doesn’t match constraints [-Werror]
  211 |             : "rm"(pred), "rm"(old_val), "i"(expected));
      |                                                        ^
trts_veh.cpp:211:56: error: impossible constraint in ‘asm’
cc1plus: all warnings being treated as errors

Reply:
Fixed in a force-push. The debug build was failing because this function wasn't being inlined by the compiler, and therefore `expected` must be a register.